src/API/WorkController.ts

```diff
@@ -32,17 +41,17 @@ export class WorkController {
     return this.collection.findOne({ id }, { projection: { _id: false } })
   }
 
-  async getByFilters(worksFilters: WorksFilters = {}): Promise<ReadonlyArray<any>> {
+  async getByFilters(worksFilters: WorksFilters = {}): Promise<WorksWithCount> {
     this.logger.trace({ method: 'getByFilters', worksFilters }, 'Getting Work by Filters from DB')
     const { offset, limit, ...filters } = worksFilters
-    return this.collection
+    const works = await this.collection
       .find(filters, { projection: { _id: false } })
-      .sort({
-        _id: -1,
-      })
+      .sort({ _id: -1 })
       .skip(offset)
       .limit(limit || 10)
       .toArray()
+    const count = await this.collection.count(filters)
```

Comment:
I'm wondering if this hits the database twice?
Maybe we could just get the results of find(), use length of that to get count
and then do works = results.sort().skip().limit().toArray()?

Reply:
sort(), skip(), limit(), and toArray() are all mongodb methods, so I'm not sure that we could do it this way.

I tried and we can do it like this: 
```ts
const results = await this.collection
      .find(filters, { projection: { _id: false } })
    const resultsArray = await results.toArray()
    const count = resultsArray.length
    const works = await results
      .sort({ _id: -1 })
      .skip(offset)
      .limit(limit || 10)
      .toArray()
```
I don't think that would be any better and might be hitting the database 3 times?

Reply:
`.count(filters)` was incorrect though. I updated it to `this.collection.find(filters, { projection: { _id: false } }).count()`